vm_supervisor/models.py

```diff
@@ -187,16 +190,27 @@ def start_watching_for_updates(self, pubsub: PubSub):
             )
 
     async def watch_for_updates(self, pubsub: PubSub):
-        await pubsub.msubscribe(
-            self.original.code.ref,
-            self.original.runtime.ref,
-            self.original.data.ref if self.original.data else None,
-            *(
-                volume.ref
-                for volume in (self.original.volumes or [])
-                if hasattr(volume, "ref")
-            ),
-        )
+        if self.is_instance:
+            await pubsub.msubscribe(
+                self.original.rootfs.parent,
+                self.original.data.ref if hasattr(self.original, "data") and self.original.data else None,
+                *(
+                    volume.ref
+                    for volume in (self.original.volumes or [])
+                    if hasattr(volume, "ref")
+                ),
+            )
+        else:
+            await pubsub.msubscribe(
+                self.original.code.ref,
+                self.original.runtime.ref,
+                self.original.data.ref if self.original.data else None,
+                *(
+                    volume.ref
+                    for volume in (self.original.volumes or [])
+                    if hasattr(volume, "ref")
+                ),
+            )
         logger.debug("Update received, stopping VM...")
         await self.stop()
```

Comment:
Do we really want to stop instances on volume updates? I'm not sure the whole concept of updating the volumes of an instance makes sense. To be discussed with @hoh.

Reply:
Interesting question. 
- Immutable volumes: could be updated in place, ignored or updated after a reboot.
- Persistent volumes: it could be handy allow resizing (size increase).

If an instance uses a volume from a third party, that third party would be able to initiate the reboot of the instances, which may be problematic.

When users use `use_latest = True` on volumes, they expect updates to the volumes to be applied to running functions. We could keep the same behavior for instances but put that field by default to `False`, hence mitigating the security vulnerability.

Reply:
I only left the subscription for volumes, to cover the use case that @hoh comments.